Update getRootTreeNodeAsync return type

alexgomezlf opened the pull request:
- getRootTreeNodeAsync should throw if the rootTreeNode doesn't exist, not return undefined

Commit:
Update getRootTreeNodeAsync return type, should throw if rootNode doesn't exist, not return undefined

Commit:
Add logic to catch error thrown by root node

paolo-arge-lf commented on projects/ui-components/lf-repository-browser/ILfTreeNodeService.ts:
```diff
@@ -15,28 +15,28 @@ export interface LfTreeNodePage {
 export interface LfTreeNodeService {
```
/**
A tree data structure provider that can be used as a data source for UI components, such as the lf-repository-browser component,  to navigate a tree structure.
*/

Commit:
add comment for tree service

paolo-arge-lf commented on projects/ui-components/lf-repository-browser/ILfTreeNodeService.ts:
```diff
@@ -1,42 +1,44 @@
+ */
 export interface LfTreeNodeService {
+  /**
+   * Gets LfTreeNode data for the lf-repository-browser to display
```
Returns a page of children for the parent folder. The implementation may provide expose ordering and filtering functionality.

paolo-arge-lf commented on projects/ui-components/lf-repository-browser/ILfTreeNodeService.ts:
```diff
@@ -1,42 +1,44 @@
+  /**
+   * Gets LfTreeNode data for the lf-repository-browser to display
+   * @param folder: string that represents the ID of the folder to get data from
+   * @param nextPage: string representing the next page requested
```
@param nextPage: string representing the next page requested. If undefined, the first page is returned.

Commit:
update getFolderChildrenAsync tsdoc

Commit:
update console error message if error getting root node

Commit:
update documentation

Commit:
simplify initAsync interface to only accept LfTreeNode
remove getTreeNodeByIdAsync from LfTreeNodeService interface

Commit:
update documentation and tests for repository-browser

Commit:
fix lint

Commit:
fix initAsync test to have error when getRootTreeNodeAsync throws

alexgomezlf commented on projects/lf-documentation/src/app/lf-repository-browser-documentation/demo-repo-service.ts:
```diff
@@ -304,14 +304,18 @@ export class DemoRepoService implements LfTreeNodeService {
+    const path = entry.path.split('//');
+    path.pop(); // remove self
+    const parent = path.pop();
+    const parentId = parent ? this._entries[parent] : this._rootEntry;
```
change name

alexgomezlf commented on projects/lf-documentation/src/app/lf-repository-browser-documentation/lf-repository-browser-documentation.component.html:
```diff
@@ -113,7 +113,7 @@
-    <td>(providers: LfTreeNodeService, selectedEntryId?: string) => Promise&lt;void&gt;</td>
+    <td>(treeNodeService: LfTreeNodeService, selectedEntry?: LfTreeNode) => Promise&lt;void&gt;</td>
     <td>
       initializes the lf-repository-browser with a LfRepositoryProvider You <b>must</b> call this method. Each time this is called, the
```
update description

alexgomezlf commented on projects/ui-components/lf-repository-browser/lf-repository-browser.component.ts:
```diff
@@ -340,14 +344,15 @@ export class LfRepositoryBrowserComponent implements OnDestroy {
-    if (rootEntry == null) {
-      console.error(`Repository browser does not contain a root entry`);
+    try {
+      const rootEntry: LfTreeNode = await this.treeNodeService.getRootTreeNodeAsync();
```
remove initializeWithRootOpne

alexgomezlf commented on projects/ui-components/lf-repository-browser/lf-repository-browser.component.ts:
```diff
@@ -279,27 +282,28 @@ export class LfRepositoryBrowserComponent implements OnDestroy {
     if (currentEntry && !currentEntry.isContainer) {
       currentEntry = await this.treeNodeService.getParentTreeNodeAsync(currentEntry);
     }
+    if (currentEntry?.id == null) {
```
!currentEntry

Commit:
update demo service to use '/' separator
update documentation initAsync description
update null checks in repository-browser component

Commit:
remove unnecessary test for initAsync with id

alexgomezlf committed:
Merge branch '13.x' into alex/update-tree-node-service-interface
